mobile/src/store/courses/actions.ts

```diff
@@ -48,6 +58,94 @@ const addCourse = createAsyncThunk<
   return course;
 });
 
+const getMentorsByCourseId = createAsyncThunk<
+  UserDetailsResponseDto[],
+  CourseGetMentorsRequestDto,
+  AsyncThunkConfig
+>(ActionType.GET_MENTORS, async (payload, { extra }) => {
+  const { coursesApi } = extra;
+  const mentors = await coursesApi.getMentorsByCourseId(payload);
+
+  return mentors;
+});
+
+const updateVisibilityBecomeMentor = createAsyncThunk<
+  boolean,
+  void,
+  AsyncThunkConfig
+>(ActionType.UPDATE_VISIBILITY_BECOME_MENTOR, async (_, { getState }) => {
+  const {
+    auth: { user },
+    courses: { mentors, course },
+  } = getState();
+  const isMentorBecomingVisible =
+    course?.courseCategoryId &&
+    !mentors.some((mentor) => mentor.id === user?.id);
+
+  return Boolean(isMentorBecomingVisible);
+});
+
+const setBecomeMentorInvisible = createAsyncThunk<
+  boolean,
+  void,
+  AsyncThunkConfig
+>(ActionType.SET_BECOME_MENTOR_INVISIBLE, async () => {
+  return false;
+});
+
+const createMentor = createAsyncThunk<
+  CoursesToMentorsResponseDto,
+  CoursesToMentorsRequestDto,
+  AsyncThunkConfig
+>(ActionType.CREATE_MENTOR, async (payload, { extra }) => {
+  const { mentorsApi } = extra;
+
+  return mentorsApi.create(payload);
+});
+
+const becomeMentor = createAsyncThunk<void, void, AsyncThunkConfig>(
+  ActionType.BECOME_MENTOR,
+  async (_, { dispatch, getState, extra }) => {
+    const {
+      courses: { course },
+      auth: { user },
+    } = getState();
+
+    const { interviewersApi } = extra;
+
+    if (!user || !course) {
+      return;
+    }
+
+    const passedCategoryInterviews =
+      await interviewersApi.getPassedInterviewCategoryIds(user.id);
+    const isCategoryPassed = passedCategoryInterviews.includes(
+      course.courseCategoryId,
+    );
+
+    if (isCategoryPassed) {
+      const payload: CoursesToMentorsRequestDto = {
+        userId: user.id,
+        courseId: course.id,
+      };
+      await dispatch(createMentor(payload));
```

Comment:
If it's failed, we need to fail this action too
```suggestion
      await dispatch(createMentor(payload)).unwrap();
```